src/pages/domain/BaseVerifyDomainPage.tsx

```diff
@@ -39,26 +36,32 @@ function OrderedListRow({index, children}: PropsWithChildren<{index: number}>) {
     );
 }
 
-type VerifyDomainPageProps = PlatformStackScreenProps<WorkspacesDomainModalNavigatorParamList, typeof SCREENS.WORKSPACES_VERIFY_DOMAIN>;
+type BaseVerifyDomainPageProps = {
+    /** The accountID of the domain */
+    accountID: number;
 
-function VerifyDomainPage({route}: VerifyDomainPageProps) {
+    /** Route to navigate to after successful verification */
+    forwardTo: Route;
+};
+
+function BaseVerifyDomainPage({accountID, forwardTo}: BaseVerifyDomainPageProps) {
     const styles = useThemeStyles();
     const theme = useTheme();
     const {translate} = useLocalize();
 
-    const accountID = route.params.accountID;
     const [domain, domainMetadata] = useOnyx(`${ONYXKEYS.COLLECTION.DOMAIN}${accountID}`, {canBeMissing: true});
     const domainName = domain ? Str.extractEmailDomain(domain.email) : '';
     const {isOffline} = useNetwork();
-
+    const [adminAccess] = useOnyx(ONYXKEYS.COLLECTION.SHARED_NVP_PRIVATE_ADMIN_ACCESS, {canBeMissing: false});
```

Comment:
Can you add this to a selector? We dont need the adminAccess for anything else so we could already just get the isAdmin value using the selector